### PR TITLE
KB: Fee Elimination Article

### PR DESCRIPTION
**Pull request by dylanireland**
Added an article that addresses **Fee Elimination** on Casper 2.0 (Condor). The article discusses gas, how it was spent on Casper 1.x, and the changes made to the gas economy in Casper 2.0. It looks at potential dangers, their avoidances, and the promising possibilities of such a system.

### CODE REVIEW

File: articles/005-fee-elimination.md [hunk 1]
```diff
@@ -0,0 +1,125 @@
+# Fee Elimination on Casper 2.0
+
+The Casper 2.0 (aka Condor) network upgrade introduces new options to the way a Casper Network can be configured to handle gas consumption. Scheduled along with the release of Condor into Mainnet is a change in the configuration of the Casper Network to use a model known as **Fee Elimination** for gas payments. The purpose of this article is to introduce this model, and describe how Fee Elimination will affect the behaviour of the Casper Network once Condor is released.
```
Comment (melpadden) [inline articles/005-fee-elimination.md:3]: "Behaviour" : we should probably decide to use American English across the board right? My bad.

File: articles/005-fee-elimination.md [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+* **Refund**: All or a portion of the remaining token after gas is purchased for execution.
+
+> [!NOTE]
+> The Casper node software supports a number of configurable options which govern how gas may be calculated for a given transaction. A discussion of these is outside the scope of this article. This article is concerned with how these gas costs are dealt with, once calculated. Gas cost options will be the subject of another article.
+
+## Fee Elimination
+
+> __Fee Elimination is the strategy of placing temporary holds on transactor balances corresponding to their incurred gas costs, instead of taking those costs from their on-chain balances__.
+
+Under 1.x, transactors must pay for gas directly from their purse balances. With Fee Elimination on Casper 2.0, a hold is placed on the calculated **Gas Cost** for a configurable period of time known as the **Hold Period**. Fees are therefore not forfeighted by transactors, and funds are not spent to execute transactions.
```
Comment (melpadden) [inline articles/005-fee-elimination.md:24]:
```suggestion
Under 1.x, transactors must pay for gas directly from their purse balances. With Fee Elimination on Casper 2.0, a hold is placed on the calculated **Gas Cost** for a configurable period of time known as the **Hold Period**. Fees are therefore not forfeited by transactors, and funds are not spent to execute transactions.
```

File: articles/005-fee-elimination.md (hunk 1 above)
Comment (melpadden) [inline articles/005-fee-elimination.md:3]:
```suggestion
The Casper 2.0 (aka Condor) network upgrade introduces new options to the way a Casper Network can be configured to handle gas consumption. Scheduled along with the release of Condor into Mainnet is a change in the configuration of the Casper Network to use a model known as **Fee Elimination** for gas payments. The purpose of this article is to introduce this model, and describe how Fee Elimination will affect the behavior of the Casper Network once Condor is released.
```

File: articles/005-fee-elimination.md [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+
+However, any gas mechanism that doesn't charge users could be vulnerable to denial-of-service attacks. Provided a large enough bankroll, a user could deploy enough transactions to slow the network for the amount of time needed for his or her previous gas payments to unlock, and use these unlocked funds to deploy more transactions, and thus repeat the process ad infinitum. In this way, one could theoretically deploy infinite transactions, cycling through their locked and unlocked balances. 
+
+Attacking the network in this way is a challenge of economic feasibility, much like many other aspects of blockchains. To prevent an attack like this from taking place, it must be made prohibitively expensive to mount such an attack, with little to no incentive to the attacker. Casper's approach involves using a long locking period, combined with 16 second blocktimes. The Casper 2.0 mainnet is slated to roll out with a 30 day locking period, but if increased, the cost to attack would scale linearly.
```
Comment (melpadden) [inline articles/005-fee-elimination.md:84]: I would add a note here to emphasise that this 30-day hold period is just for example, and is not a certainty. The last conversation I had about this talked about a 90 day hold period. it is liable to change until close to the final rollout.

File: articles/005-fee-elimination.md [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+
+## Looking Forward
+
+By introducing Fee Elimination to the Casper Network, we hope to make transacting with the blockchain more accessible and less financially cumbersome. With this new model, interacting with smart contracts can become effectively free for users, inviting larger audiences to participate in new and exciting protocols.
```
Comment (melpadden) [inline articles/005-fee-elimination.md:109]: "new and exciting protocols".... other protocols? Not Casper? Not sure what you mean here. I would leave that last clause out, it feels a bit too broad.

File: articles/005-fee-elimination.md [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+#### Proposer
+A validator proposing a block to the network for execution  
+[Consensus](https://docs.casper.network/concepts/economics/consensus/)  
+[Validator](https://docs.casper.network/concepts/glossary/V/#validator)
+
+#### Fees
+A portion of a transaction's gas costs given over to the proposer of the block containing that transaction.  
+[Gas Concepts](https://docs.casper.network/concepts/economics/gas-concepts/)  
+[Runtime Economics](https://docs.casper.network/runtime/)  
+
```
Comment (melpadden) [inline articles/005-fee-elimination.md:125]:
```suggestion
[Validator Rewards](https://docs.casper.network/concepts/glossary/R/#reward)
```

File: articles/005-fee-elimination.md (hunk 4 above)
Comment (dylanireland) [reply articles/005-fee-elimination.md:109]: It refers to smart contract-based protocols, that run on Casper. Will rephrase.

**Review by melpadden — APPROVED** [review]

**Pull request merged**